evalml/tests/component_tests/test_arima_regressor.py

```diff
@@ -145,6 +146,26 @@ def test_set_forecast(get_ts_X_y):
     assert fh_.is_relative
 
 
+def test_get_sp():
+    X = pd.DataFrame({"dates": pd.date_range("2021-01-01", periods=500, freq="D")})
+    X.ww.init()
+    clf_day = ARIMARegressor(time_index="dates")
+    sp_ = clf_day._get_sp(X)
+    assert sp_ == 7
+
+    X = pd.DataFrame({"dates": pd.date_range("2021-01-01", periods=500, freq="M")})
+    X.ww.init()
+    clf_month = ARIMARegressor(time_index="dates")
+    sp_ = clf_month._get_sp(X)
+    assert sp_ == 12
+
+    X = pd.DataFrame({"dates": pd.date_range("2021-01-01", periods=500, freq="2D")})
```

Comment:
I think going back to one of my earlier comments, if X already knows the frequency, we might not want to have the freq re-infered.  Perhaps we want to add a test counting the number of times `infer_temporal_freqs()` is called in the case of a known frequency provided and limit it to 1?